CMake: `AMReX::Flags_FASTMATH` Interface Helper Target

ax3l opened the pull request:
## Summary

This PR _changes no defaults_. It just refactors #4545 a bit further prior to adding the option `AMReX_FASTMATH` in a first AMReX release, providing a better internal implementation and a user-friendly helper target (fully optional).

Add an interface target for the fast-math flags. This makes it easier to re-use them downstream when AMReX is pre-compiled w/o fast-math (default in all of our current package managers), but the main code wants to still use fast-math on the side of the application code.

## Additional background

Follow-up to #4545, see comment https://github.com/AMReX-Codes/amrex/pull/4545#discussion_r2245683394

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [x] include documentation in the code and/or rst files, if appropriate

ax3l committed:
CMake: AMReX_FASTMATH Interface Target

Add an interface target for the FASTMATH flags.

WeiqunZhang commented:
This changes the default for the CUDA backend. Previously, fastmath is not on by default for C++ flags, although it's on for CUDA flags. It is now on by default in both flags for CUDA backend.

ax3l commented on Tools/CMake/AMReX_Config.cmake:
```diff
@@ -147,19 +149,7 @@ function (configure_amrex AMREX_TARGET)
-           # See https://cmake.org/cmake/help/v4.1/variable/CMAKE_LANG_COMPILER_ID.html#variable:CMAKE_%3CLANG%3E_COMPILER_ID
-           target_compile_options(${AMREX_TARGET} PUBLIC
-               $<$<CXX_COMPILER_ID:AppleClang,Clang,CrayClang,GNU,IBMClang,IntelLLVM,XLClang>:-ffast-math>
-               $<${_cxx_msvc}:"/fp:fast">  # MSVC
```
Just realized this also fixes a bug. The original code caused a compile error on MSVC because of the quotes
```
       (ClCompile target) -> 
         \"\fp:fast"(1,1): error C1083: Cannot open source file: '"/fp:fast"': No such file or directory [D:\a\impactx\impactx\build\_deps\fetchedamrex-build\Src\amrex_3d.vcxproj]
```
https://github.com/BLAST-ImpactX/impactx/pull/1073

ax3l commented:
> This changes the default for the CUDA backend. Previously, fastmath is not on by default for C++ flags, although it's on for CUDA flags. It is now on by default in both flags for CUDA backend.

Ouch!

From Slack:

You can see it in the pr's github action. https://github.com/AMReX-Codes/amrex/actions/runs/16654514074/job/47135937738#step:5:108

`-ffast-math` is there

but not on development branch's ci: https://github.com/AMReX-Codes/amrex/actions/runs/16655180798/job/47138252299#step:5:108

ax3l committed:
Fix: `AMReX_FASTMATH` default OFF

That one controls the backend in full now, CUDA can for now just
overwrite it with `AMReX_CUDA_FASTMATH` (ON by default).

ax3l commented:
Thank you for catching that, fix added!
